### PR TITLE
add TalentCastEfficiencyRequirement component

### PR DESCRIPTION
**Pull request by ToppleTheNun**
add CombatLogParserContext to make CombatLogParser available
via hooks for functional components.

add TalentCastEfficiencyRequirement component
that will only render if the selected combatant
has the given talent.

TalentCastEfficiencyRequirement will also fetch
the cast efficiency of the given talent so that
we don't need to set up an AbilityRequirement in
each spec checklist.

update Vengeance checklist to use it as an
example.

screenshot of it in use:
image

### CODE REVIEW

File: src/interface/report/CombatLogParserContext.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  <CombatLogParserCtx.Provider value={{ combatLogParser }}>{children}</CombatLogParserCtx.Provider>
+);
+
+export const useCombatLogParser = () => {
+  const context = useContext(CombatLogParserCtx);
+  if (context === undefined) {
+    throw new Error('useCombatLogParser must be within CombatLogParserProvider');
+  }
+  return context;
+};
```
Comment (emallson) [inline src/interface/report/CombatLogParserContext.tsx]: we may want to move the `Guide`'s `useAnalyzer` hook over here. its been quite useful.

(not for this PR tho, just a note for later)

File: src/interface/report/Results/index.tsx [hunk 2]
```diff
@@ -266,9 +281,11 @@ class Results extends React.PureComponent<Props, State> {
...
 
         return (
-          <div className="container">
-            <ErrorBoundary>{tab ? tab.render() : '404 tab not found'}</ErrorBoundary>
-          </div>
+          <CombatLogParserProvider combatLogParser={parser}>
+            <div className="container">
+              <ErrorBoundary>{tab ? tab.render() : '404 tab not found'}</ErrorBoundary>
+            </div>
+          </CombatLogParserProvider>
```
Comment (emallson) [inline src/interface/report/Results/index.tsx]: rather than wrapping each return value here, we should wrap the `renderContent` call on line 463 IMO

**Review by emallson — CHANGES_REQUESTED** [review]
this looks like a huge improvement overall, but one small change for the core `Results/index.tsx` when applying the context provider

File: src/interface/report/CombatLogParserContext.tsx (hunk 1 above)
Comment (ToppleTheNun) [reply src/interface/report/CombatLogParserContext.tsx]: I have another PR where I'm adding a lot of contexts, that would probably fit very well with that PR.

**Review by emallson — APPROVED** [review]

**Pull request merged**